src/common.test.ts

```diff
@@ -55,4 +55,72 @@ describe('common methods', () => {
 			expect(isValid).to.be.false;
 		});
 	});
+
+	const mockDateTimestamp = 1651785959180;
+	const aDayInMilliseconds = 1000 * 60 * 60 * 24;
+	const aWeekInMilliseconds = aDayInMilliseconds * 7;
+	const sameDay = [new Date(mockDateTimestamp), new Date(mockDateTimestamp + 1000)];
+	const differentDay = [new Date(mockDateTimestamp), new Date(mockDateTimestamp + 2 * aDayInMilliseconds)];
+	const differentWeek = [new Date(mockDateTimestamp), new Date(mockDateTimestamp + 2 * aWeekInMilliseconds)];
+	describe('daysDiffInEST function', () => {
+		it('return 0 when both dates are in the same EST day', () => {
+			const [prev, curr] = sameDay;
+			expect(daysDiffInEST(prev, curr)).to.equal(0);
+		});
+
+		it('return 2 when the dates have two days of difference in EST time', () => {
+			const [prev, curr] = differentDay;
+			expect(daysDiffInEST(prev, curr)).to.equal(2);
+		});
+	});
+
+	describe('weeksDiffInEST function', () => {
+		it('return 0 when both dates are in the same EST week', () => {
+			const [prev, curr] = sameDay;
+			expect(weeksDiffInEST(prev, curr)).to.equal(0);
+		});
+
+		it('return 2 when the dates have two weeks of difference in EST time', () => {
+			const [prev, curr] = differentWeek;
+			expect(weeksDiffInEST(prev, curr)).to.equal(2);
+		});
+	});
+
+	describe('dateToEST function', () => {
+		before(() => {
+			// TODO: set a custom timezone (e.g. UTC) in this environment
+		});
+
+		it('returns a date with 4hs behind UTC');
+	});
+
+	describe('dateToSunday method', () => {
+		it('returns a date in the first second of this week (sunday)', () => {
+			const dateOnThursday = new Date(mockDateTimestamp);
+			const dateOnPrevSunday = dateToSunday(dateOnThursday);
+			const diffInMilliseconds = dateOnThursday.getTime() - dateOnPrevSunday.getTime();
+
+			expect(diffInMilliseconds).to.be.lessThan(aWeekInMilliseconds);
+			expect(dateOnPrevSunday.getDay()).to.equal(0);
+			expect(dateOnPrevSunday.getHours()).to.equal(0);
+			expect(dateOnPrevSunday.getMinutes()).to.equal(0);
+			expect(dateOnPrevSunday.getSeconds()).to.equal(0);
+			expect(dateOnPrevSunday.getMilliseconds()).to.equal(0);
+		});
+
+		it('returns a date in the same day if already on sunday', () => {
```

Comment:
Getting this test failure in my env:

```shell
  1) common methods
       dateToSunday method
         returns a date in the same day if already on sunday:

      AssertionError: expected 597600003 to be at most 86400000
      + expected - actual

      -597600003
      +86400000
```

Reply:
We are now using a specific timezone to be deterministic with the expectations
17920ab7d22fd3f7b75e5d83e27d1e80ccb9acec

Reply:
Nice catch! :sunglasses: